test/SlowTests/Client/TimeSeries/Query/TimeSeriesLinqQuery.cs

```diff
@@ -0,0 +1,459 @@
+﻿using System;
+using System.Linq;
+using FastTests;
+using Raven.Client.Documents.Queries;
+using Raven.Client.Documents.Linq;
+using Raven.Tests.Core.Utils.Entities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Client.TimeSeries.Query
+{
+    public class TimeSeriesLinqQuery : RavenTestBase
+    {
+        public TimeSeriesLinqQuery(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private class Person
+        {
+            public string Name { get; set; }
+
+            public int Age { get; set; }
+
+            public string WorksAt { get; set; }
+        }
+
+        private class Watch
+        {
+            public string Manufacturer { get; set; }
+
+            public double Accuracy { get; set; }
+
+            //public AdditionalData AdditionalData { get; set; }
+
+            public long Min { get; set; }
+
+            public long Max { get; set; }
+
+            public bool IsoCompliant;
+
+            public DateTime EndOfWarranty { get; set; }
+
+        }
+
+        [Fact]
+        public void CanQueryTimeSeriesAggregation_UsingLinq()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var baseline = DateTime.Today;
+
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new User
+                    {
+                        Name = "Oren",
+                        Age = 35
+                    }, "users/ayende");
+
+                    session.TimeSeriesFor("users/ayende")
+                        .Append("Heartrate", baseline.AddMinutes(61), "watches/fitbit", new[] { 59d });
+                    session.TimeSeriesFor("users/ayende")
+                        .Append("Heartrate", baseline.AddMinutes(62), "watches/fitbit", new[] { 79d });
+
+                    session.TimeSeriesFor("users/ayende")
+                        .Append("Heartrate", baseline.AddMinutes(63), "watches/fitbit", new[] { 69d });
+
+                    session.TimeSeriesFor("users/ayende")
+                        .Append("Heartrate", baseline.AddMonths(1).AddMinutes(61), "watches/fitbit", new[] { 159d });
+                    session.TimeSeriesFor("users/ayende")
+                        .Append("Heartrate", baseline.AddMonths(1).AddMinutes(62), "watches/fitbit", new[] { 179d });
+
+                    session.TimeSeriesFor("users/ayende")
+                        .Append("Heartrate", baseline.AddMonths(1).AddMinutes(63), "watches/fitbit", new[] { 169d });
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Query<User>()
+                        .Where(u => u.Age > 21)
+                        .Select(u => RavenQuery.TimeSeries(u, "Heartrate")
+                            .Where(ts => ts.Tag == "watches/fitbit")
+                            .GroupBy("1 month")
```

Comment:
I really want to be able to pass strongly typed value here.
`TimeRange.Months(1)` ?

Reply:
Builder pattern would be best IMO e.g. `.GroupBy(g => g.Months(1))`

Reply:
Oh, I really like it